queue-manager/core/src/manager.ts

```diff
@@ -558,14 +595,14 @@ class Manager {
    * Usually we call this method after a change detected in the state of manager.
    *
    */
-  private handleUpdate(queue_id: QueueID) {
+  private async handleUpdate(queue_id: QueueID) {
```

Comment:
Why you need to change the behaviour of this method on the whole file? the idea was triggering `persistor` to `persist` and update data like a background job. You changed the whole logic, it will wait for persistor to sync data with storage then runs the remaining stuff. This is not the purpose of this implementation.
Do you have any requirements to change the behaviour?